packages/@aws-cdk/aws-amplify/lib/app.ts

```diff
@@ -0,0 +1,287 @@
+    super(scope, id, props);
+
+    if (props.basicAuthConfig) {
+      this.setBasicAuth(props.basicAuthConfig.username, props.basicAuthConfig.password);
+    }
+
+    if (props.environmentVariables && props.environmentVariables.length > 0) {
+      this.environmentVariablesResolver.addEnvironmentVariables(...props.environmentVariables);
+    }
+
+    const resource = new CfnApp(scope, 'App', {
+      accessToken: props.accessToken,
+      basicAuthConfig: this.basicAuthResolver,
+      buildSpec: props.buildSpec,
+      customRules: props.customRules,
+      description: props.description,
+      environmentVariables: this.environmentVariablesResolver,
+      name: props.name,
+      oauthToken: props.oauthToken,
+      repository: props.repository,
+      tags: props.tags
+    });
+
+    if (props.iamServiceRole) {
+      resource.iamServiceRole = props.iamServiceRole.roleArn;
+    }
+
+    if (props.buildSpec && !props.iamServiceRole) {
+      // create lazy role
+      const role = new LazyRole(scope, 'lazy-role', {
+        assumedBy: new ServicePrincipal('amplify.amazonaws.com')
+      });
+
+      resource.iamServiceRole = role.roleArn;
+    }
+
+    this.appArn = resource.attrArn,
+    this.appDefaultDomain = resource.attrDefaultDomain;
+    this.appId = resource.attrAppId;
+    this.appName = resource.attrAppName;
+
+    this.resource = resource;
+  }
+
+  public validate(): string[] {
+    const errors: string[] = [];
+
+    if (this.resource.buildSpec && !this.resource.iamServiceRole) {
+      errors.push(`You need to specify a service role when using buildspecs`);
+    }
+
+    return errors;
+  }
+
+  /**
+   * Add Service Role
+   *
+   * @param role
+   */
+  public addServiceRole(role: IRole) {
+    this.resource.iamServiceRole = role.roleArn;
+  }
+
+  /**
+   * Set Basic Auth on App
+   *
+   * @param username
+   * @param password
+   */
+  public setBasicAuth(username: string, password: string) {
+    this.basicAuthResolver.basicAuthConfig({
+      enableBasicAuth: true,
+      password,
+      username
+    });
+  }
+
+  /**
+   * Add Environment Variable to App
+   *
+   * @param name
+   * @param value
+   */
+  public addEnvironmentVariable(name: string, value: string) {
+    this.environmentVariablesResolver.addEnvironmentVariable(name, value);
+  }
+}
+
+/**
+ * App Interface
+ */
+export interface IApp extends IResource {
+  /**
+   * App ID
+   *
+   * @attribute
+   */
+  readonly appId: string;
+}
+
+/**
+ * App Options
+ */
+export interface AppOptions extends ResourceProps {
+  /**
+   * App Name
+   *
+   * @todo This might cause an error, it conflicts with the construct
+   *
+   * @default
+   */
+  readonly appName?: string;
+}
+
+/**
+ * App Properties
+ */
+export interface AppProps extends AppOptions {
+  /**
+   * Access Token
+   *
+   * @default
+   */
+  readonly accessToken?: string;
+
+  /**
+   * Basic Auth Config
+   *
+   * @default
+   */
+  readonly basicAuthConfig?: BasicAuthConfig;
```

Comment:
We discourage the Config suffix. Everything is config.

Reply:
It's the name of the actual property in the resource: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-amplify-app.html#cfn-amplify-app-basicauthconfig

Reply:
Still... we don't have any requirement/tenet to have 1:1 mapping between CDK APIs and CFN APIs. We omit the "Config" suffix because it doesn't add any additional information. `basicAuth` should be just fine.